subprojects/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/precompiled/PrecompiledScriptPlugins.kt

```diff
@@ -74,4 +74,4 @@ val Project.sourceSets
 
 private
 val SourceSet.kotlin: SourceDirectorySet
-    get() = @Suppress("deprecation") withConvention(KotlinSourceSet::class) { kotlin }
+    get() = @Suppress("deprecation") withConvention(KotlinSourceSet::class) { kotlin } // TODO do we have Extensions we have for this
```

Comment:
Yes, this could be changed to:
```suggestion
    get() = the<KotlinSourceSet>().kotlin
```